metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/pattern_handler.py

```diff
@@ -959,28 +961,50 @@ def create_lineage(
         )
 
         if not connect_string:
-            logger.warning(
-                "Can not extract ODBC connect string from data access function"
+            self.reporter.warning(
+                title="Can not extract ODBC connect string",
+                message="Can not extract ODBC connect string from data access function. Skipping Lineage creation.",
+                context=f"table-name={self.table.full_name}",
             )
             return Lineage.empty()
 
         logger.debug(f"ODBC connect string: {connect_string}")
         data_platform, powerbi_platform = extract_platform(connect_string)
         server_name = extract_server(connect_string)
 
+        if not data_platform:
+            dsn = extract_dsn(connect_string)
+            if dsn:
+                logger.debug(f"Extracted DSN: {dsn}")
+                server_name = dsn
+            if dsn and self.config.dsn_to_platform_name:
+                logger.debug(f"Attempting to map DSN {dsn} to platform")
+                name = self.config.dsn_to_platform_name.get(dsn)
+                if name:
+                    logger.debug(f"Found DSN {dsn} mapped to platform {name}")
+                    data_platform, powerbi_platform = normalize_platform_name(name)
+
         if not data_platform or not powerbi_platform:
-            logger.warning(
-                "Can not determine platform from ODBC connect string. Skipping Lineage creation."
+            self.reporter.warning(
+                title="Can not determine ODBC platform",
+                message="Can not determine platform from ODBC connect string. Skipping Lineage creation.",
+                context=f"table-name={self.table.full_name}",
```

Comment:
```suggestion
                context=f"table-name={self.table.full_name}, connect-string={connect_string}",
```

this may be useful to quickly spot the issue

Reply:
Added